src/Compilers/CSharp/Test/Symbol/Symbols/Source/FileModifierTests.cs

```diff
@@ -45,5 +46,737 @@ class Outer
+            verify();
+
+            void verify()
+            {
+                comp.VerifyDiagnostics();
+
+                var cs = comp.GetMembers("C");
+                Assert.Equal(2, cs.Length);
+                Assert.Equal(comp.SyntaxTrees[0], cs[0].DeclaringSyntaxReferences.Single().SyntaxTree);
+                Assert.Equal(comp.SyntaxTrees[1], cs[1].DeclaringSyntaxReferences.Single().SyntaxTree);
+            }
+        }
+
+        [Fact]
+        public void Duplication_02()
+        {
+            // As a sanity check, demonstrate that non-file classes with the same name across different files are disallowed.
+            var source1 = """
+                using System;
+
+                class C
+                {
+                    public static void M()
+                    {
+                        Console.Write(1);
+                    }
+                }
+                """;
+
+            var source2 = """
+                using System;
+
+                class C
+                {
+                    public static void M()
+                    {
+                        Console.Write(2);
+                    }
+                }
+                """;
+
+            var main = """
+
+                class Program
+                {
+                    static void Main()
+                    {
+                        C.M();
+                    }
+                }
+                """;
+
+            var comp = CreateCompilation(new[] { source1 + main, source2 });
+            verify();
+
+            comp = CreateCompilation(new[] { source1, source2 + main });
+            verify();
+
+            void verify()
+            {
+                comp.VerifyDiagnostics(
+                    // (3,7): error CS0101: The namespace '<global namespace>' already contains a definition for 'C'
+                    // class C
+                    Diagnostic(ErrorCode.ERR_DuplicateNameInNS, "C").WithArguments("C", "<global namespace>").WithLocation(3, 7),
+                    // (5,24): error CS0111: Type 'C' already defines a member called 'M' with the same parameter types
+                    //     public static void M()
+                    Diagnostic(ErrorCode.ERR_MemberAlreadyExists, "M").WithArguments("M", "C").WithLocation(5, 24),
+                    // (14,11): error CS0121: The call is ambiguous between the following methods or properties: 'C.M()' and 'C.M()'
+                    //         C.M();
+                    Diagnostic(ErrorCode.ERR_AmbigCall, "M").WithArguments("C.M()", "C.M()").WithLocation(14, 11));
+
+                var cs = comp.GetMember("C");
+                var syntaxReferences = cs.DeclaringSyntaxReferences;
+                Assert.Equal(2, syntaxReferences.Length);
+                Assert.Equal(comp.SyntaxTrees[0], syntaxReferences[0].SyntaxTree);
+                Assert.Equal(comp.SyntaxTrees[1], syntaxReferences[1].SyntaxTree);
+            }
+        }
```

Comment:
>} [](http://example.com/codeflow?start=8&length=1)

Are we testing duplicate names with `partial` and `file`? For instance: { `partial class C { }`, `partial class C { }`, `file class C { }` } in combinations across one or more files.

Reply:
Found a bug with a new test related to this, thanks.